packages/dd-trace/src/profiling/encoders/pprof.js

```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+const util = require('util')
+const zlib = require('zlib')
+const gzip = util.promisify(zlib.gzip)
+const { perftools } = require('../../../../../proto/profile')
+const { Profile } = perftools.profiles
+
+class Encoder {
+  async encode (profile) {
```

Comment:
The `async` keyword here is unnecessary, since `gzip` returns a promise.

Reply:
That's not entirely true since this code could throw so it would still have to be wrapped in a promise if the caller expects a promise.

Reply:
`Profile.encode()` can throw? That's fine, because all the callers are already within async functions (without their own try/catch blocks) so it will propagate the same way, as a rejection.

Reply:
The contract of a method can only assume if it should return a promise or not, but not how the caller will use the promise. You can thank ECMAScript 😄

Reply:
Isn't this internal though? If you control all the usages, you control both sides of the contract, and you can avoid unnecessary promise creation.

Reply:
Designing internal APIs as if they were is good practice and allows to more easily expose internals.

Reply:
My primary concern here is the extra promise allocation in the non-error path.

This could be solved with the following:

```
encode (profile) {
  let encoded
  try {
    encoded = Profile.encode(profile).finish()
  } catch (e) {
    return Promise.reject(e)
  }
  return gzip(encoded)
}
```

This way the function still always returns a promise, does not throw, and doesn't allocate any more promises than needed to maintain those constraints.

It does slightly sacrifice cleanliness, so you can decide whether it's worth it or not.

Reply:
Since this is not a hot path at all and the gain or removing a single promise is very minimal, I would be inclined to keep the current approach and cleaner code.

Reply:
SGTM.